user accounts manipulation audit example (#8158)
opensearch-project/documentation-website: _security/audit-logs/index.md

To disable audit logs after they've been enabled, remove the `plugins.security.audit.type: internal_opensearch` setting from `opensearch.yml`, or switch off the **Enable audit logging** check box in OpenSearch Dashboards.

## Audit user account manipulation

> To enable audit logging on changes to the security index (i.e. changes to roles mappings, creation/deletion of roles, etc.), use the following settings in the `compliance:` portion of the audit log configuration.

Reviewer: ```suggestion
To enable audit logging on changes to a security index, such as changes to roles mappings and role creation or deletion, use the following settings in the `compliance:` portion of the audit log configuration, as shown in the following example:

```

To disable audit logs after they've been enabled, remove the `plugins.security.audit.type: internal_opensearch` setting from `opensearch.yml`, or switch off the **Enable audit logging** check box in OpenSearch Dashboards.

## Audit user account manipulation

> By default, audit of user account creation/removal is off (`AUTHENTICATED` audit events disabled). To enable it, issue a similar request:

Reviewer 2: I don't think AUTHENTICATED is the right event here. An AUTHENTICATED even is audit logged every time a request is sent with valid credentials.

To monitor changes to the security index, I think you would use the `compliance:` section of the audit log config. Specifically, the `write_watched_indices: []` [here](https://github.com/opensearch-project/security/blob/main/config/audit.yml#L80).

The security index is `.opendistro_security`. If you add it into this list, you should be able to track changes to internalusers, roles, roles_mappings, etc.

Author: Thanks, Craig. I've just tried to put `AUTHENTICATED` event back to the list of the disabled categories and added the `.opendistro_security` index to the `write_watched_indices` list. And it didn't seem to be enough to get the details like the created or deleted user account name. To get that details, I had to disable `write_metadata_only` and enable `write_log_diffs` options as well.
Please, confirm this is the right way to go and I'll update my PR

Reviewer 2: Yes, that's correct. Does the section below look correct to you? Is there anything else that needs to be added?

```
  compliance:
    # enable/disable compliance
    enabled: true

    # Log updates to internal security changes
    internal_config: true

    # Log only metadata of the document for write events
    write_metadata_only: false

    # Log only diffs for document updates
    write_log_diffs: true

    # List of indices to watch for write events. Wildcard patterns are supported
    # write_watched_indices: ["twitter", "logs-*"]
    write_watched_indices: [".opendistro_security"]
```

Author: yes, looks just fine, thank you! i'd just add a comment regarding the specified `.opendistro_security` index to make people aware why it's added by default

Reviewer 2: Yes please update this PR. Thank you @jazzl0ver ! This will be a great addition to the documentation.

Author: @cwperks updated

Reviewer 2: wdyt including the entire section from the [above comment](https://github.com/opensearch-project/documentation-website/pull/8158#discussion_r1743900351)?

In general, the audit logs documentation on the documentation website looks like it could be updated to describe all potential settings in the `audit.yml` file in a table and the `compliance` portion of the audit log needs to be expanded upon.

For this PR, I think including the blurb from above would be an improvement on the existing docs.

Author: done
